pkg/network/http/telemetry.go

```diff
@@ -27,10 +26,10 @@ type telemetry struct {
 	rejected                                    *atomic.Int64 `stats:""` // this happens when an user-defined reject-filter matches a request
 	malformed                                   *atomic.Int64 `stats:""` // this happens when the request doesn't have the expected format
 	aggregations                                *atomic.Int64 `stats:""`
-
-	reporter stats.Reporter
 }
 
+var telemetryReporter = atomicstats.NewReporter((*telemetry)(nil))
```

Comment:
Is a global necessary?

Reply:
It's an un-exported global, at least!  It represents the type, but with some runtime analysis via `reflect` since we can't do it at build time.  So there's definitely only one instance.

Reply:
Yeah, it is more that importing any part of the package will initialize and run this reflection, even if the reported type is never used.

Reply:
Potentially fixable with a `sync.Once` -- but is that worth it?

Reply:
Anything preventing the reporter from being a struct member and initialized in `New` functions, like it was before?

Reply:
I realize it's minor, but I'm trying to build a package that's easy to use. Embedding the reporter in the instance only makes sense when the instance is a singleton.  Initializing a global on instance initialization is potentially racy.  So I guess by "is that worth it" I mean, either (a) is it worth trying to explain that in the README and potentially having other users of this utility mess it up; or (b) is it worth keeping the README as-is but deviating from it in this case, to save a few ns at startup?

Reply:
We/our = network-agent team